### PR TITLE
poc: [M3-7799] - Improve Docs Site Sidebar

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Updates the docs site sidebar to be generated based on file names and folder structure of the `docs/` folder 🗂️
- I think this will be nice because it will enable anyone to add documentation without needing to modify any Vitepress configuration 🔧
- Updates Vitepress to latest 🆕

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-02-21 at 3 46 12 PM | Screenshot 2024-02-21 at 4 45 18 PM  |

## How to test 🧪

- Check out this PR
- Create a test markdown file at `docs/development-guide/example/testing.md` with some markdown in it
- Run `yarn docs`
- Go to `http://localhost:5173/manager/`
- Observe that the new docs page was added to the sidebar with correct nesting 📂

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: docs/.vitepress/tsconfig.json
Comment (bnussman-akamai) [inline docs/.vitepress/tsconfig.json:1]: Had to create this so I could call `replaceAll` in `docs/.vitepress/plugins/sidebar.ts`

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -15,12 +17,7 @@ export default {
     search: {
       provider: "local",
     },
-    sidebar: [
-      {
-        text: "Development Guide",
-        items: guides,
-      },
-    ],
```
Comment (bnussman-akamai) [inline docs/.vitepress/config.ts:23]: Instead of having to do this manually, it will be done automatically based on the folder/file names 🎉🗂️

**Review by abailly-akamai — APPROVED** [review]
Nice changes! 

confirmed dynamic sidebar works as expected and build succeeds ✅

**Review by mjac0bs — APPROVED** [review]
Works great, thank you!

Screenshot 2024-02-22 at 9 45 45 AM

**Comment by bnussman-akamai** [comment]
@mjac0bs What you sent in the screenshot doesn't look quite right. 🤔

The title in the sidebar should be `My Test Section` based on the fact you named the file `15-my-test-section.md`

You'll have to restart the VitePress server for the sidebar to regenerate when you checkout my PR.

**Pull request merged**